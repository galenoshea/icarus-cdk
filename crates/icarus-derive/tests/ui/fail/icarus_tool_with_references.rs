use icarus_derive::*;

#[icarus_tool("Tool with reference parameters")]
fn bad_tool(data: &str) -> Result<String, String> {
    Ok(data.to_string())
}

fn main() {}