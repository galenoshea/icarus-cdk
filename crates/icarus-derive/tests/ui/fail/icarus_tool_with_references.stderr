error: Tool function validation failed:
         - Tool functions must have either #[query] or #[update] attribute
         - Tool parameters cannot contain references. Use owned types instead.
 --> tests/ui/fail/icarus_tool_with_references.rs:4:4
  |
4 | fn bad_tool(data: &str) -> Result<String, String> {
  |    ^^^^^^^^
